backend/internal/handler.go

```diff
@@ -48,7 +50,24 @@ func ParseAndValidateRequestBody(ctx context.Context, v *validator.Validate, r *
 func WriteJSONResponse(w http.ResponseWriter, status int, data interface{}) {
 	w.Header().Set("Content-Type", "application/json")
 	w.WriteHeader(status)
-	if err := json.NewEncoder(w).Encode(data); err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
+	jsonBytes, err := json.Marshal(data)
+	if err != nil {
+		http.Error(w, "Failed to marshal response", http.StatusInternalServerError)
+		return
+	}
+
+	_, err = w.Write(jsonBytes)
+	if err != nil {
+		http.Error(w, "Failed to write response", http.StatusInternalServerError)
+		return
+	}
+}
+
+func ParseUUID(value string) (uuid.UUID, error) {
+	var res uuid.UUID
+	err := res.Scan(value)
+	if err != nil {
+		return res, errorPkg.ErrInvalidUUID
```

Comment:
This function could use some improvements:
1. If `Scan()` fails, it returns an uninitialized `uuid.UUID`, which might confuse the caller. Consider returning `uuid.UUID{}` explicitly.
2. Consider using fmt.Errorf("failed to parse UUID: %w", err) to wrap the original error, preserving its context for better debugging.
3. The variable name res could be more descriptive, e.g., parsedUUID.